src/test/java/uk/gov/pay/api/service/CreatePaymentServiceTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.pay.api.service;
+
+import au.com.dius.pact.consumer.PactVerification;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.api.app.RestClientFactory;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.app.config.RestClientConfig;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.model.CreatePaymentRequest;
+import uk.gov.pay.api.model.PaymentState;
+import uk.gov.pay.api.model.TokenPaymentType;
+import uk.gov.pay.api.model.links.Link;
+import uk.gov.pay.api.model.links.PaymentWithAllLinks;
+import uk.gov.pay.api.model.links.PostLink;
+import uk.gov.pay.api.pact.PactProviderRule;
+import uk.gov.pay.api.pact.Pacts;
+
+import javax.ws.rs.client.Client;
+import java.util.Collections;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreatePaymentServiceTest {
+    private CreatePaymentService createPaymentService;
+
+    @Rule
+    public PactProviderRule connectorRule = new PactProviderRule("connector", this);
+
+    private Client client;
+
+    @Mock
+    private PublicApiConfig configuration;
+    private PublicApiUriGenerator publicApiUriGenerator;
+    private ConnectorUriGenerator connectorUriGenerator;
+    
+    @Before
+    public void setup() {
+        when(configuration.getConnectorUrl()).thenReturn(connectorRule.getUrl()); // We will actually send real requests here, which will be intercepted by pact
+
+        when(configuration.getBaseUrl()).thenReturn("http://publicapi.test.localhost/"); // Has to match what is in the Pact
```

Comment:
i'm also not sure what this comment means and how it relates to pact. if anything, the publicapi-connector.json pact file has a "self" link `http://connector.service.backend/v1/api/accounts/GATEWAY_ACCOUNT_ID/charges/ch_ab2341da231434l`.

Reply:
i think it's actually related to publicApiUriGenerator rather than pact